src/Http/Middleware/CP/RedirectIfAuthorized.php

```diff
@@ -16,8 +16,8 @@ class RedirectIfAuthorized
      */
     public function handle($request, Closure $next, $guard = null)
     {
-        if (User::current()) {
-            return redirect(cp_route('index'));
+        if ($user = User::current()) {
+            return redirect($user->can('access cp') ? cp_route('index') : '/');
```

Comment:
It might be worth showing a toast message when logged in users are redirected to the Control Panel to avoid any confusion about *why* they've been redirected away from the reset/activation page.

Reply:
Yeah, I think that should be fine. It can probably be something generic like "You can't do this while logged in".

Reply:
Cool, done